### DISCUSSION TITLE
Plausible contribution environment setup

### DISCUSSION THREAD

**Discussion by bradycargle** [plausible/analytics#1567] [Anything Else]
Hey there, how are you? :) I'm working on contributing to Plausible by fixing some bug reports.

I'm trying to set up my dev environment and running into issues. I've tried to follow the [contribution guidelines](https://github.com/plausible/analytics/blob/master/CONTRIBUTING.md)  and, when that didn't work well, following the [self-hosting](https://plausible.io/docs/self-hosting) docs. Both have been a bit of a dead end

Some questions:

1. On self-hosting, I can get Plausible up and running. But where do I find the files to contribute to? I cloned the plausible/hosting repo, but it's empty of all the plausible/analytics repo files that contain what I need. Am I missing something?

2. If I shouldn't go the self-hosting route and should instead follow the contributing guidelines, where do I run "make postgres" and "make clickhouse"? I tried running in a few directories and no luck

Thanks. I'm really looking forward to contributing :)

**Comment by m-col** [comment]
Hey @bradycargle, first you need to clone the codebase from your fork with `git clone https://github.com/bradycargle/analytics`. This will create the `analytics` folder in the current directory - it is from inside this folder (at the top level) where running `make clickhouse` and `make postgres` will start those services with docker (assuming you have that installed and running). Hope that helps!

**Reply by ukutaht** [reply]
More information here: https://github.com/plausible/analytics/blob/master/CONTRIBUTING.md
